maven/bnd-export-maven-plugin/src/main/java/aQute/bnd/maven/export/plugin/ExportMojo.java

```diff
@@ -47,6 +49,7 @@
 import biz.aQute.resolve.ResolveProcess;
 
 @Mojo(name = "export", defaultPhase = PACKAGE, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
+@SuppressWarnings("deprecation")
```

Comment:
This seems horribly wrong to suppress deprecation at the class level. What is deprecated and why would we add new code which uses deprecated things? I can see that over time, things we use can become deprecated and we decide to continue to use the newly deprectated thing. But I don't think we should add new function which starts off using deprecated things. We should use the replacement for the deprecated thing or find another way which is not deprecated.

Reply:
I'll have to investigate more deeply, but it appears the Maven project deprecates things before there is a replacement. I searched the Maven codebase high and low and there is no alternative for this deprecated API. Even Maven itself still uses it.

Reply:
If we must use the deprecated API, then we need to minimize the scope of the deprecation suppression and not use it on a broad scope such as the class. Using it on a broad scope will hide other deprecation warnings we might want to do something about.

Reply:
I can do that.